src/main/java/com/feywild/feywild/block/FeyMushroomBlock.java

```diff
@@ -0,0 +1,95 @@
+import io.github.noeppi_noeppi.libx.base.BlockBase;
+import io.github.noeppi_noeppi.libx.mod.ModX;
+import net.minecraft.client.renderer.ItemBlockRenderTypes;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.core.BlockPos;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.block.BonemealableBlock;
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.VoxelShape;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+
+import javax.annotation.Nonnull;
+import java.util.Random;
+import java.util.function.Consumer;
+
+public class FeyMushroomBlock extends BlockBase implements BonemealableBlock {
+
+    protected static final VoxelShape SHAPE = Block.box(5.0D, 0.0D, 5.0D, 11.0D, 6.0D, 11.0D);
+
+    public FeyMushroomBlock(ModX mod) {
+        super(mod, BlockBehaviour.Properties.copy(Blocks.RED_MUSHROOM)
+                .lightLevel((mushroom) -> mushroom.getValue(BlockStateProperties.LEVEL)));
+
+        this.registerDefaultState(this.stateDefinition.any()
+                .setValue(BlockStateProperties.LEVEL, 7));
+    }
+
+    @Override
+    @OnlyIn(Dist.CLIENT)
+    public void registerClient(ResourceLocation id, Consumer<Runnable> defer) {
+        defer.accept(() -> ItemBlockRenderTypes.setRenderLayer(this, RenderType.cutout()));
+    }
+
+    @Nonnull
+    @Override
+    @SuppressWarnings("deprecation")
+    public VoxelShape getShape(@Nonnull BlockState state, @Nonnull BlockGetter level, @Nonnull BlockPos pos, @Nonnull CollisionContext context) {
+        return SHAPE;
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> stateBuilder) {
+        stateBuilder.add(BlockStateProperties.LEVEL);
+    }
+
+    @Nonnull
+    @Override
+    @SuppressWarnings("deprecation")
+    public InteractionResult use(@Nonnull BlockState state, @Nonnull Level level, @Nonnull BlockPos pos, @Nonnull Player player, @Nonnull InteractionHand hand, @Nonnull BlockHitResult hit) {
+        if (player.getItemInHand(hand).getItem() == ModItems.feyDust) {
+            if (!level.isClientSide) {
+                state = state.cycle(BlockStateProperties.LEVEL);
```

Comment:
This reassigns the method parameter, you should avoid this. Also I think the flags should be 3 so it causes a block update. Just inline it to `level.setBlock(pos, state.cycle(BlockStateProperties.LEVEL), 3)`.